chromium_src/chrome/browser/ui/webui/tab_search/tab_search_page_handler.cc

```diff
@@ -0,0 +1,255 @@
+/* Copyright (c) 2025 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#include "chrome/browser/ui/webui/tab_search/tab_search_page_handler.h"
+
+#include "base/check.h"
+#include "brave/browser/ai_chat/ai_chat_service_factory.h"
+#include "brave/components/ai_chat/core/browser/ai_chat_service.h"
+#include "brave/components/ai_chat/core/browser/constants.h"
+#include "brave/components/ai_chat/core/browser/engine/engine_consumer.h"
+#include "brave/components/ai_chat/core/browser/types.h"
+#include "brave/components/ai_chat/core/common/features.h"
+#include "brave/components/ai_chat/core/common/mojom/ai_chat.mojom.h"
+#include "brave/components/ai_chat/core/common/pref_names.h"
+#include "chrome/browser/ui/browser_finder.h"
+#include "chrome/browser/ui/browser_navigator_params.h"
+#include "chrome/browser/ui/tabs/tab_strip_model.h"
+#include "components/grit/brave_components_strings.h"
+#include "components/sessions/core/session_id.h"
+#include "ui/base/l10n/l10n_util.h"
+#include "ui/base/page_transition_types.h"
+#include "ui/base/window_open_disposition.h"
+#include "url/origin.h"
+
+#define TabSearchPageHandler TabSearchPageHandler_ChromiumImpl
+#include "src/chrome/browser/ui/webui/tab_search/tab_search_page_handler.cc"
+#undef TabSearchPageHandler
+
+TabSearchPageHandler::TabSearchPageHandler(
+    mojo::PendingReceiver<tab_search::mojom::PageHandler> receiver,
+    mojo::PendingRemote<tab_search::mojom::Page> page,
+    content::WebUI* web_ui,
+    TopChromeWebUIController* webui_controller,
+    MetricsReporter* metrics_reporter)
+    : TabSearchPageHandler_ChromiumImpl(std::move(receiver),
+                                        std::move(page),
+                                        web_ui,
+                                        webui_controller,
+                                        metrics_reporter) {
+  pref_change_registrar_.Add(
+      ai_chat::prefs::kBraveAIChatTabOrganizationEnabled,
+      base::BindRepeating(
+          &TabSearchPageHandler::OnTabOrganizationFeaturePrefChanged,
+          base::Unretained(this), Profile::FromWebUI(web_ui_)));
+}
+
+void TabSearchPageHandler::OnTabOrganizationFeaturePrefChanged(
+    Profile* profile) {
+  page_->TabOrganizationEnabledChanged(
+      ai_chat::AIChatServiceFactory::GetForBrowserContext(profile) &&
+      ai_chat::features::IsTabOrganizationEnabled() &&
+      profile->GetPrefs()->GetBoolean(
+          ai_chat::prefs::kBraveAIChatTabOrganizationEnabled));
+}
+
+TabSearchPageHandler::~TabSearchPageHandler() = default;
+
+ai_chat::EngineConsumer* TabSearchPageHandler::GetAIEngineForTabOrganization() {
+  if (!ai_chat_engine_) {
```

Comment:
So we don't use the `ai_chat_engine_` that the user has picked in the `AiChat` sidebar?

Reply:
Correct, it's independent from user choices due to model capabilities.